app/src/main/java/de/westnordost/streetcomplete/quests/memorial_type/AddMemorialType.kt

```diff
@@ -0,0 +1,45 @@
+package de.westnordost.streetcomplete.quests.memorial_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement
+
+class AddMemorialType : OsmFilterQuestType<String>() {
```

Comment:
Please create an enum class and/or sealed interface for the answer. Maybe something like this:

```kt
sealed interface MemorialTypeAnswer

enum class MemorialType(val osmValue: String) : MemorialTypeAnswer {
    STATUE("statue"),
    BUST("bust"),
    PLAQUE("plaque"),
    WAR_MEMORIAL("war_memorial"),
    STONE("stone"),
    OBELISK("obelisk"),
}

enum class Stele(val osmMaterialValue: String) : MemorialTypeAnswer {
    WOODEN_STELE("wood"),
    STONE_STELE("stone"),
    KHACHKAR_STELE("stone"),
}
```

Then `applyAnswerTo` can be simplified to:

```kt
when (answer) {
    is MemorialType -> {
        tags["memorial"] = answer.osmValue
    }
    is Stele -> {
        tags["memorial"] = "stele"
        tags["material"] = answer.osmMaterialValue
    }
}

if (answer == KHACHKAR_STELE) {
    tags["stele"] = "khachkar"
}
```